LDM-633.tex

```diff
@@ -0,0 +1,319 @@
+should be executed (e.g. to postpone execution of steps which are likely to
+fail towards the end of a pipeline or start processing from specific steps).
+
+\usecase{BPS}{Divide a pipeline into smaller subpipelines}
+The Operator divides a pipeline into smaller subpipelines. (e.g. when the
+actual number of outputs generated by the pipeline upstream determines the size
+of a subpipeline downstream).
+
+\usecase{BPS}{Group pipelines steps into a pipeline jobs}
+The Operator groups subsets of pipeline steps together into smaller number of
+compute jobs to reduce the time overhead related to job startup costs.
+
+\usecase{BPS}{Customize a pipeline}
+The Operator needs the ability to customize a pipeline for various reasons including:
+\begin{itemize}
+  \item 
+    only need to run the first steps in a larger pipeline to produce outputs
+    needed for debugging.
+  \item
+    run last steps from a larger pipeline starting with outputs of previous
+    executions of the larger pipeline.
+  \item
+    set the order in which steps constituting the pipeline should be executed
+    (e.g. to postpone execution of steps which are likely to fail towards the
+    end of a pipeline).
+\end{itemize}
+
+\section{Execution}
+
+\usecase{BPS}{Initiate a campaign}
+The Operator specifies a set of pipelines, their configurations, set of inputs,
+an initial priority (either for the entire campaign or for individual
+pipelines), and submits them for processing.
+
+\usecase{BPS}{Terminate a failed campaign/pipelines}
+The Operator declares a running campaign/pipelines to be a failure and wants to terminate the processing before its completion.
+
+\usecase{BPS}{Pause/Unpause a campaign/pipelines}
+The Operator or CM wants to pause dispatching of new compute jobs or pipelines
+in a campaign to release the resources, e.g., for a campaign with higher
+priority or for maintenance, allowing currently running compute jobs or
+pipelines to continue running.  The operator will also need to unpause
+campaigns/pipelines that have been put on hold.  This does not include the
+ability to change configuration, etc. for the campaign/pipeline.
+
+\usecase{BPS}{Prioritize campaign/pipelines}
+The Operator wants to designate different processing priorities for
+campaigns/pipelines at time of submission or needs to alter those priorities
+for pending pipelines.
+
+\usecase{BPS}{Dispatch campaign/pipeline/jobs to specific platform(s)}
+The Operator or the Campaign Manager dispatches campaign/pipeline/jobs to
+compute resources where the computational platform(s) may be specified and may
+provide a list of specific machine(s) to avoid (e.g., run on cluster at
+CC-IN2P3 but avoid specific machines cc3000 and cc4003).
+
+\usecase{BPS}{Execute a campaign on different architectures}
+The Operator wants the flexibility to execute pipelines on platforms with
+different architectures (e.g. with and without shared filesystem).
+
+\usecase{BPS}{Run a pipeline using Gen3 Middleware}
```

Comment:
Does this mean the payload is required to use the Gen3 Middleware?  How about the Gen2/as-is pipeline framework?

Reply:
Yes, the framework will support running Gen2 Middleware too until it will be completely deprecated. But you're right, the document doesn't say it explicitly. I'll add the relevant use case.